File: src/main/scala/com/vc/easy/L26.scala
package com.vc.easy

object L26 {
  def removeDuplicates(arr: Array[Int]): Int = {
    val n = arr.length
    if(n == 0) return 0
    var lastUniqueIndex = 0
    var i = 0
    while(i < n - 1) {
      if(arr(i) != arr(i + 1)) {
        arr(lastUniqueIndex + 1) = arr(i + 1)
        lastUniqueIndex += 1
      }
      i += 1
    }
    lastUniqueIndex + 1
  }
}

File: src/main/java/com/vc/easy/L26.java
package com.vc.easy;

class L26 {
    public int removeDuplicates(int[] nums) {
        int prev = 0, current = 0;
        while(current < nums.length) {
            if(nums[prev] == nums[current]) {
                current++;
            }
            else {
                prev++;
                nums[prev] = nums[current++];
            }
        }
        return prev + 1;
    }
}
